docs/textsearch.adoc

```diff
@@ -182,4 +182,19 @@ mgmt.buildIndex('name', Vertex.class).addKey(string, Mapping.STRING.asParameter(
 mgmt.buildIndex('document', Vertex.class).addKey(text, Mapping.TEXT.asParameter()).buildMixedIndex("search")
 mgmt.commit()
 
-With these settings, JanusGraph will use a SimpleAnalyzer analyzer for property key 'string', an EnglishAnalyzer analyzer for property key 'text',  a KeywordAnalyzer analyzer for property 'name' and a StandardAnalyzer analyzer for property 'document'.
\ No newline at end of file
+With these settings, JanusGraph will use a SimpleAnalyzer analyzer for property key 'string', an EnglishAnalyzer analyzer for property key 'text',  a KeywordAnalyzer analyzer for property 'name' and a StandardAnalyzer analyzer for property 'document'.
+
+==== Custom parameters
+
+Sometimes it is required to set additional parameters on mappings (other than mapping type, mapping name and analyzer). For example, when we would like to use a different similarity algorithm (to modify the scoring algorithm of full text search) or if we want to use a custom boosting on some fields in Elasticsearch we can set custom parameters (right now only Elasticsearch supports custom parameters).
+The name of the custom parameter must be set through `ParameterType.customParameterName("yourProperty")`.
+
+===== For Elasticsearch
+
+[source, gremlin]
+mgmt = graph.openManagement()
+myProperty = mgmt.makePropertyKey('my_property').dataType(String.class).make()
+mgmt.buildIndex('custom_property_test', Vertex.class).addKey(myProperty, Mapping.TEXT.asParameter(), Parameter.of(ParameterType.customParameterName("boost"), 5.5), Parameter.of(ParameterType.customParameterName("similarity"), "DFR")).buildMixedIndex("search")
```

Comment:
It's not an issue with this PR but I got an error `Datatype is not supported by database since no serializer has been registered: class java.math.BigDecimal` with this example. I had to change the boost to 5 (integer) to get it to work for testing.

Also I got an error trying to use DFR similarity: `Unknown Similarity type [DFR] for field`. I changed to "boolean" to get it to work for testing. From [docs](https://www.elastic.co/guide/en/elasticsearch/reference/current/similarity.html) it looks like using DFR requires additional index settings configuration as described in [similarity module ](https://www.elastic.co/guide/en/elasticsearch/reference/current/index-modules-similarity.html) section.